agent/grpc/public/services/serverdiscovery/watch_servers.go

```diff
@@ -0,0 +1,157 @@
+package serverdiscovery
+
+import (
+	"context"
+	"errors"
+
+	"github.com/hashicorp/consul/acl"
+	"github.com/hashicorp/consul/agent/consul/autopilotevents"
+	"github.com/hashicorp/consul/agent/consul/stream"
+	"github.com/hashicorp/consul/agent/grpc/public"
+	"github.com/hashicorp/consul/agent/structs"
+	"github.com/hashicorp/consul/proto-public/pbserverdiscovery"
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/go-uuid"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// We tag logs with a unique identifier to ease debugging. In the future this
+// should probably be an Open Telemetry trace ID.
+func streamID() string {
+	id, err := uuid.GenerateUUID()
+	if err != nil {
+		return ""
+	}
+	return id
+}
+
+// WatchServers provides a stream on which you can receive the list of servers
+// that are ready to receive incoming requests including stale queries. The
+// current set of ready servers are sent immediately at the start of the
+// stream and new updates will be sent whenver the set of ready servers changes.
+func (s *Server) WatchServers(req *pbserverdiscovery.WatchServersRequest, serverStream pbserverdiscovery.ServerDiscoveryService_WatchServersServer) error {
+	logger := s.Logger.Named("watch-servers").With("trace_id", streamID())
```

Comment:
Looks like we use a different identifier for each the public grpc request (`trace_id`, `request_id`, `stream_id`). I think when we add the middleware/interceptor, we could make it consistent.

Reply:
Absolutely we should. For now though I can at least unify them to request_id.

Reply:
All of the endpoints should now be using `request_id`